File: src/main/java/com/adamkali/simpleide/editor/StatusBar.java
package com.adamkali.simpleide.editor;

import com.adamkali.simpleide.Global;

import javax.swing.*;
import java.awt.*;

public class StatusBar extends JComponent {
    public static final int STATUS_BAR_HEIGHT = 32;
    private static final int MARGIN_LEFT = 8;
    private static final int MARGIN_TOP = 4;
    private static final int LINE_HEIGHT = 20;

    @Override
    protected void paintComponent(Graphics g) {
        super.paintComponent(g);
        g.setColor(Color.LIGHT_GRAY);

        g.fillRect(0, 0, getWidth(), getHeight());
        g.setColor(Color.BLACK);
        g.drawString("SimpleIDE", MARGIN_LEFT, MARGIN_TOP + LINE_HEIGHT / 2);
        g.drawString(String.format("Line: %d, Column: %d", Global.getCursor().getCurrentLine() + 1, Global.getCursor().getCurrentColumn() + 1), getWidth() - 200, MARGIN_TOP + LINE_HEIGHT / 2);
    }
}

File: src/main/java/com/adamkali/simpleide/editor/StatusBar.kt
package com.adamkali.simpleide.editor

import com.adamkali.simpleide.Global
import com.adamkali.simpleide.activity.CursorActivityListener
import com.adamkali.simpleide.editor.io.TextPosition
import com.adamkali.simpleide.preferences.EditorColors
import java.awt.Dimension
import java.awt.Graphics
import javax.swing.JComponent

/**
 * StatusBar is a component that displays
 * the current line and column number of the cursor
 */
class StatusBar : JComponent(), CursorActivityListener {
    /** The left margin of the StatusBar  */
    val MARGIN_LEFT = 8

    init {
        font = Global.getFont()
        isOpaque = true
        background = EditorColors.statusBarBackground()
        preferredSize = Dimension(0, STATUS_BAR_HEIGHT)
        minimumSize = Dimension(0, STATUS_BAR_HEIGHT)
        Global.getCursor().setActionListener(this)
    }

    /**
     * Paints the StatusBar
     * @param g the Graphics object to paint on
     */
    override fun paintComponent(g: Graphics?) {
        super.paintComponent(g)
        if (g == null) return

        g.color = EditorColors.statusBarBackground()
        g.fillRect(0, 0, width, height)

        g.font = font
        val fm = g.fontMetrics
        val textY = (height + fm.ascent - fm.descent) / 2
        g.color = EditorColors.statusBarForeground()
        g.drawString("SimpleIDE", MARGIN_LEFT, textY)

        val position = formatCursorPosition(Global.getCursor().getLine(), Global.getCursor().getColumn())
        val positionWidth = fm.stringWidth(position)
        g.drawString(position, width - positionWidth - MARGIN_LEFT, textY)
    }

    companion object {
        @JvmField
        var STATUS_BAR_HEIGHT: Int = 22

        @JvmStatic
        fun formatCursorPosition(line: Int, column: Int): String {
            return String.format("Ln %d, Col %d", line + 1, column + 1)
        }
    }

    override fun onCursorMove(from: TextPosition, to: TextPosition) {
        repaint()
    }
}
